### PR TITLE
now doesnt fail when encountering a new element in groups

### PR DESCRIPTION
**Pull request by JAicewizard**
I can't create issues, so it wouls be nice if all elements would be supported, like polygon etc

### CODE REVIEW

**Comment by Shugyousha** [comment]
Hi

Thanks for the pull request!

[2018-12-24 05:11] JAicewizard <notifications@github.com>
> I can't create issues, so it woule be nice if all elements would be
> supported, like polygon etc

Currently I cannot test these changes since I am out of the office. I
will do so once I am back from the holidays.

There are some stylistic changes in these commits that I don't think
are worth it though. I will point those out separately.

File: svg.go [hunk 1]
```diff
@@ -212,11 +213,11 @@ func (s *Svg) UnmarshalXML(decoder *xml.Decoder, start xml.StartElement) error {
 
 			switch tok.Name.Local {
 			case "g":
-				g := &Group{Owner: s, Transform: mt.NewTransform()}
```
Comment (Shugyousha) [inline svg.go]: I would prefer not to change this. I don't think there is any benefit to this change.

File: svg.go [hunk 2]
```diff
@@ -231,6 +232,8 @@ func (s *Svg) UnmarshalXML(decoder *xml.Decoder, start xml.StartElement) error {
 
 			if err = decoder.DecodeElement(dip, &tok); err != nil {
 				return fmt.Errorf("error decoding element of SVG struct: %s", err)
+			} else {
```
Comment (Shugyousha) [inline svg.go]: There seems to be something missing here...

File: svg.go [hunk 3]
```diff
@@ -245,7 +248,7 @@ func (s *Svg) UnmarshalXML(decoder *xml.Decoder, start xml.StartElement) error {
 
 // ParseSvg parses an SVG string into an SVG struct
 func ParseSvg(str string, name string, scale float64) (*Svg, error) {
-	var svg Svg
+	svg := Svg{}
```
Comment (Shugyousha) [inline svg.go]: I would prefer to leave these as they are. Why do you want to change these?

**Comment by JAicewizard** [comment]
sorry for the unintentional changes, I added them while figuring out why it didnt work. it should be back to how it was now

**Comment by Shugyousha** [comment]
I have merged your patches into the dev branch (squashing the last commit). As soon as our testing is done, I will merge dev into master. Thanks for the patches!

**Pull request closed without merging**